Added mail on final run option + test

aidanheerdegen opened the pull request:
Closes #299 

Adds a config option to mail user when last run of a sequence completes.

aidanheerdegen committed:
Added mail on final run option + test

pep8speaks commented:
Hello @aidanheerdegen! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`payu/schedulers/pbs.py`](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py):

> [Line 100:67](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L100): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 101:25](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L101): [E127](https://duckduckgo.com/?q=pep8%20E127) continuation line over-indented for visual indent
> [Line 101:54](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L101): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 108:21](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L108): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 110:29](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L110): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 110:31](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L110): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 111:31](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L111): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 111:33](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/payu/schedulers/pbs.py#L111): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals

* In the file [`test/test_pbs.py`](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/test/test_pbs.py):

> [Line 171:9](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/test/test_pbs.py#L171): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 171:21](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/test/test_pbs.py#L171): [E201](https://duckduckgo.com/?q=pep8%20E201) whitespace after '('
> [Line 171:62](https://github.com/payu-org/payu/blob/84703c815571086f4a50ea6b5c74dec485e3dd11/test/test_pbs.py#L171): [E202](https://duckduckgo.com/?q=pep8%20E202) whitespace before ')'

aidanheerdegen committed:
PEP8 fixes

aidanheerdegen committed:
Set  mail on final default to False.

Changed logic for final run to remove unecessary inclusion of the
current run number.

Fixed bug with email_address variable name.

aidanheerdegen committed:
Fixed test to work with mail option not defaulting to true

navidcy commented:
Can't really comment on the python implementation of this, but *I applaud* the feature.
